.github/workflows/docker.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and Publish Docker Image
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-publish:
+    env:
+      SHOULD_PUSH_IMAGE: ${{ (github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev')) || github.event_name == 'workflow_dispatch' }}
```

Comment:
For what it's worth btw, some repos MAY be on a `main` branch. That isn't the case here, but it should be noted for the other PRs. I'm assuming you're going to just copy-paste from this one, rather than hand-writing them each time, so might be worth just accounting for it everywhere

Reply:
Our PR generation script handles main branches, same with repos that require buildstring arg in the dockerfile

Reply:
Btw there are 16 uses of "main" branch, there are 11 uses of "master" branch amongst the nex based game server repos

Reply:
I was not aware you had a script for this :+1: sounds good